src/main/java/uk/gov/justice/digital/delius/controller/AuthenticationController.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.justice.digital.delius.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import uk.gov.justice.digital.delius.data.api.UserDetails;
+import uk.gov.justice.digital.delius.service.UserService;
+
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.http.HttpStatus.OK;
+
+@RestController
+public class AuthenticationController {
+
+    private final UserService userService;
+
+    public AuthenticationController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @RequestMapping("/authenticate")
+    public ResponseEntity authenticate(final @RequestParam String username, @RequestParam String password) {
+        boolean authenticated = userService.authenticateUser(username, password);
+        if(!authenticated) {
+            return new ResponseEntity(HttpStatus.UNAUTHORIZED);
+        }
+        return new ResponseEntity(HttpStatus.OK);
+    }
+
+    @RequestMapping(value = "/users/{username}/details", method = RequestMethod.GET)
+    public ResponseEntity<UserDetails> findUser(final @PathVariable("username") String username) {
+        return userService.getUserDetails(username).map(userDetails -> new ResponseEntity<>(userDetails, OK))
+                .orElse(new ResponseEntity<>(NOT_FOUND));
+
+    }
+
+    @RequestMapping(value = "/users/{username}/password", method = RequestMethod.POST)
```

Comment:
Should there be a safeguard against any call to the API from being able to lock/unlock/change passwords for any user? just wondering if the API should be protected to specific users/roles?  I know its currently in a locked down environment, but without authorisation on the caller could any client could use these endpoints?

Reply:
Yes they could right now. And until the API is protected this is exactly how it works now. Any client can masquerade as any user without a password. There is essentially no security other than this is in UKCloud. I did think about adding some sort of protection just for this APIs but I would be wasting my time since I can still access all data anyway with no credentials.

My gut feeling is we will eventually protect these APIs with a service to service token (issued by AUTH) and everything else will be the user's token (also issued by AUTH). But by definition we have no AUTH server right now so we can't.

One final thing to note; in live there is no password in LDAP it is in active directory so this is all post-aws migration work so we can't do any damage in live. The APIs essentially only work in test environments. Hence it is still a spike that can have no side-affect in live even if someone to ssh on to a UKCloud box or got through your NDH tunnel.